### ISSUE TITLE
Training Model leads to Coco Accuracy -1.000

### ISSUE THREAD

**Issue by paviddavid** [tensorflow/models#6688]
### System information
- **What is the top-level directory of the model you are using**: /tensorflow/models/object_detection/
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes (to create tfrecord files, see above)
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 16.04
- **TensorFlow installed from (source or binary)**: N/A
- **TensorFlow version (use command below)**: 1.12.0-rc2
- **Bazel version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: 10.0
- **GPU model and memory**: GeForce GTX 980 Ti 6GB
- **Exact command to reproduce**: See Below

### Describe the problem
I am using a pre-trained Faster_RCNN_Resnet101 model that is trained on Kitti image data and want to fine-tune this model with Cityscapes data. The accuracy during the training process is -1.000 the whole time and does not change. Seems like the Coco training accuracy.. ?

Command to reproduce the training process:
 
```
python object_detection/model_main.py \
    --pipeline_config_path={Path_To_Config_File} \
    --model_dir={Path_To_Pretrained_Model} \
    --num_train_steps={500000} \
    --sample_1_of_n_eval_examples= {1}\
    --alsologtostderr
```

Output after each training iteration:
![grafik](https://user-images.githubusercontent.com/45428272/56869214-e15b2500-69fd-11e9-9ffd-40c7f643a0dc.png)

Why does the accuracy look so weird? Why is the accuracy over the whole training process always -1.000? 

I really appreciate every hint/remark/solution on this solution. Does anybody know why the accuracy is looking so weird and does not output "normal" accuracy values?

Thanks a lot in advance

### Source code / logs
Information containing in my custom Cityscapes tfRecord file:
```
    tf_example = tf.train.Example(features=tf.train.Features(feature={
    'image/height': dataset_util.int64_feature(height),
    'image/width': dataset_util.int64_feature(width),
    'image/filename': dataset_util.bytes_feature(filename),
    'image/source_id': dataset_util.bytes_feature(filename),
    'image/encoded': dataset_util.bytes_feature(encoded_image_data),
    'image/format': dataset_util.bytes_feature(image_format.encode('utf8')),
    'image/object/bbox/xmin': dataset_util.float_list_feature(xmins),
    'image/object/bbox/xmax': dataset_util.float_list_feature(xmaxs),
    'image/object/bbox/ymin': dataset_util.float_list_feature(ymins),
    'image/object/bbox/ymax': dataset_util.float_list_feature(ymaxs),
    'image/object/difficult': dataset_util.int64_list_feature(difficult_obj),
    'image/object/class/text': dataset_util.bytes_list_feature(classes_text),
    'image/object/class/label': dataset_util.int64_list_feature(classes),
    }))
```

**Comment by JinlongLi2016** [comment]
The same problem!
1. Some people mentioned that it's probably caused by mislabeled annotations or broken images. 
  Checked. Mine seems to be fine.

**Comment by paviddavid** [comment]
@JinlongLi2016 Which data do you use? Which pre-trained model? Which data for fine-tuning?
I am using a faster R-CNN Resnet101 model that is trained on Kitti data and want to fine-tune it with Cityscapes data.

Til now, I did not figure out how it works. Changing the `metrics_set `was also not helpful. 
What have you tried until now? Maybe we can figure it out together what went wrong. 
Any ideas?

**Comment by JinlongLi2016** [comment]
@paviddavid  Own dataset. I use ssdlite-mobilenetv1 and faster-rcnn (and corrsponding pretrained model from official site.  Maybe I should try 'public dataset' to check whether this problem will occur.

I didn't use coco metrics. In fact, i didn't specify evaluation metric. It should be the default evaluation metric: [something related to pascal](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/installation.md#coco-api-installation).

**Comment by paviddavid** [comment]
@JinlongLi2016 Weird behavior.. 
So you generate your own tfrecord file? Have you checked if they contain all necessary information? 
Some people mentioned that this could be the cause of the error. 

I suggest to 1) check the tfrecord file and 2) the config file.

**Comment by JinlongLi2016** [comment]
@paviddavid  Thanks for advice.  1) I have all the info except `difficult`.   2) config file was copied from [official repo](https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_mobilenet_v1_pets.config). I modified `num_classes`,  `train_input_reader`  'eval_input_reader'  `input_path` fields and label_map_path.  `fine_tune_checkpoint` field was modified also.

**Comment by JinlongLi2016** [comment]
Also, when I load pretrained model, it logs that the BatchNorm parameters alphas and betas' momentum are not available in checkpoint. I don't know whether this will be a potential problem with small batch size.

**Comment by paviddavid** [comment]
@JinlongLi2016 OK, so it is less the same what I have done. I read fine-tuning requires a re-initialization of the last layer of your network but I do not know if this could be a reason. In many tutorials, the only steps they provide are the ones I described. 

For your second comment, I do not know what could be the source of the error. You downloaded the config file, so there should be everything correct. Seems like something is missing from the checkpoint. Have you re-downloaded the model, maybe something went wrong within in the download?

Let me know, when you solved the issue.

**Comment by JinlongLi2016** [comment]
@paviddavid  I transferred from windows to linux,  and train my network with download VGGpets , it works fine up dto ~2k steps. 
When I load my small dataset( just a portion from the whole dataset), program runs fine, performance not checked.
When i load all dataset, it blows up from ~0.5k steps. I think  some problem exits in my data.

As for your problem with opendataset, I think you might change some setting. If you don't mind, can you share you config file with me. Mine is [here](<script src="https://gist.github.com/JinlongLi2016/35275bcd2a89ef3a5f1bc4c81d85ad62.js"></script>).

**Comment by paviddavid** [comment]
@JinlongLi2016 What happens after ~2k steps? 
So, as i understood right, everything works fine using VGGpets dataset but when you change to your custom dataset it does not work anymore?

I attached my config file. Did you change something in your config file aside from the paths that should be configured?


```
# Faster R-CNN with Resnet-101 (v1)
# Trained on KITTI dataset (cars and pedestrian), initialized from COCO
# detection checkpoint.
# Users should configure the fine_tune_checkpoint field in the train config as
# well as the label_map_path and input_path fields in the train_input_reader and
# eval_input_reader. Search for "PATH_TO_BE_CONFIGURED" to find the fields that
# should be configured.

model {
  faster_rcnn {
    num_classes: 2
    image_resizer {
      keep_aspect_ratio_resizer {
        # Raw KITTI images have a resolution of 1242x375, if we wish to resize
        # them to have a height of 600 then their width should be
        # 1242/(375/600)=1987.2
        min_dimension: 600
        max_dimension: 1987
      }
    }
    feature_extractor {
      type: 'faster_rcnn_resnet101'
      first_stage_features_stride: 16
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 14
    maxpool_kernel_size: 2
    maxpool_stride: 2
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: false
        dropout_keep_probability: 0.5
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.0
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 300
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
  }
}

train_config: {
  batch_size: 1
  optimizer {
    momentum_optimizer: {
      learning_rate: {
        manual_step_learning_rate {
          initial_learning_rate: 0.0001
          schedule {
            step: 500000
            learning_rate: .00001
          }
          schedule {
            step: 700000
            learning_rate: .000001
          }
        }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  gradient_clipping_by_norm: 10.0
  fine_tune_checkpoint: "/home/user/tensorflow/models/research/object_detection/data/faster_rcnn_resnet101_kitti_2018_01_28/model.ckpt" #PATH_TO_BE_CONFIGURED
  from_detection_checkpoint: true
  num_steps: 800000
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
}

train_input_reader: {
  label_map_path: "/home/user/tensorflow/models/research/object_detection/data/cityscapes_label_map.pbtxt" #PATH_TO_BE_CONFIGURED
  tf_record_input_reader: {
    input_path: "/home/user/tensorflow/models/research/object_detection/data/Cityscapes_REDUCED_Man_TRAINING.tfrecord" #PATH_TO_BE_CONFIGURED
  }
}

eval_config: {
  use_moving_averages: false
  num_examples: 500
}

eval_input_reader: {
  label_map_path: "/home/user/tensorflow/models/research/object_detection/data/kitti_label_map.pbtxt" #PATH_TO_BE_CONFIGURED
  tf_record_input_reader: {
    input_path: "/home/user/tensorflow/models/research/object_detection/data/kitti.record_val.tfrecord" #PATH_TO_BE_CONFIGURED
  }
}

```

**Comment by JinlongLi2016** [comment]
To 3k steps, loss looked normal( decrease as proceding), so I terminated it.

**Comment by paviddavid** [comment]
@JinlongLi2016 But with your custom data set it is working or not? 

Do you see any differences in the config files?

**Comment by paviddavid** [comment]
@JinlongLi2016 I figured out what went obviously wrong. I had a mismatching between the class names in the label map and the class labels of the tfrecord file. Upper and lower case error...

**Comment by JinlongLi2016** [comment]
...wtf, my tfrecords has file names in Chinese, some error occurs. So I change class names in label.pbtxt to integer indices....So my file names should mismatch too...

**Comment by JinlongLi2016** [comment]
@paviddavid  Great! I will check that tomorrow.

**Comment by paviddavid** [comment]
@JinlongLi2016 It could be soo easy.. But glad to see that you also figured it out, awesome!

**Comment by JinlongLi2016** [comment]
@paviddavid  Problem solved.  Thanks a lot.

**Comment by paviddavid** [comment]
@JinlongLi2016 Great!

**Comment by ymodak** [comment]
Closing this issue since its resolved. Thanks!
